packages/ws/src/strategies/sharding/WorkerShardingStrategy.ts

```diff
@@ -41,9 +42,21 @@ export enum WorkerReceivePayloadOp {
 	WorkerReady,
 }
 
+// Use a string enum so that the values can just be used as a lookup into InternalEvents
+export enum ShardEvents {
+	Close = 'Close',
+	ConnectionError = 'ConnectionError',
+	Debug = 'Debug',
+	Dispatch = 'Dispatch',
+	HeartbeatComplete = 'HeartbeatComplete',
+	Hello = 'Hello',
+	Ready = 'Ready',
+	Resumed = 'Resumed',
+}
+
 export type WorkerReceivePayload =
-	// Can't seem to get a type-safe union based off of the event, so I'm sadly leaving data as any for now
-	| { data: any; event: WebSocketShardEvents; op: WorkerReceivePayloadOp.Event; shardId: number }
+	// No way to get type safety on data
+	| { data: any; event: ShardEvents; op: WorkerReceivePayloadOp.Event }
```

Comment:
I know it was like this before, but perhaps it would be better to type `data` as `unknown`
```suggestion
	| { data: unkown; event: ShardEvents; op: WorkerReceivePayloadOp.Event }
```

Reply:
Can't. Makes me unable to `Evt#post` (or `EventEmitter#emit` in the past) w/o a ts-expect-error or some other cast on that line. Probs prefer it as-is.